code/modules/vore/fluffstuff/custom_items_vr.dm

```diff
@@ -36,34 +36,100 @@
+	if(!skip_contents && O.contents.len) //check if we're loaded/modified, in the event of gun/suit kits, to avoid purging stuff like ammo, badges, armbands, or suit helmets
+		to_chat(user, "<span class='warning'>You should probably remove any attached items or loaded ammunition before trying to modify that!</span>")
 		return
-	if(!isturf(O.loc))
-		to_chat(user, "<span class='warning'>[O] must be safely placed on the ground for modification.</span>")
+	if(cost > parts)
+		to_chat(user, "<span class='warning'>The kit doesn't have enough parts left to modify that.</span>")
+		if(can_revert && ((to_helmet_cost || to_suit_cost) < 0))
+			to_chat(user, "<span class='notice'> You can recover parts by using the kit on an already-modified item.</span>")
 		return
+	if(keycheck && owner_ckey) //check if we're supposed to care
+		if(user.ckey != owner_ckey) //ERROR: UNAUTHORIZED USER
+			to_chat(user, "<span class='warning'>You probably shouldn't mess with all these strange tools and parts...</span>") //give them a slightly fluffy explanation as to why it didn't work
+			return
 	playsound(user.loc, 'sound/items/Screwdriver.ogg', 100, 1)
-	var/N = new to_type(O.loc)
-	user.visible_message("<span class='warning'>[user] opens \the [src] and modifies \the [O] into \the [N].</span>","<span class='warning'>You open \the [src] and modify \the [O] into \the [N].</span>")
+	var/obj/N = new to_type(O.loc)
+	user.visible_message("<span class='notice'>[user] opens \the [src] and modifies \the [O] into \the [N].</span>","<span class='notice'>You open \the [src] and modify \the [O] into \the [N].</span>")
+	
+	//crude, but transfer prints and fibers to avoid forensics abuse, same as the bloody/gooey check above
+	N.fingerprints = O.fingerprints
+	N.fingerprintshidden = O.fingerprintshidden
+	N.fingerprintslast = O.fingerprintslast
+	N.suit_fibers = O.suit_fibers
+	
+	//I don't like using the lookdown checks here but fuck it, it works. the istypes are essential though, or else it runtimes and won't get to the qdel for the source item, allowing item duping
+	//transfer logic could technically be made more thorough and handle stuff like helmet/boots/tank vars for suits, but in those cases you should be removing the items first anyway
+	if(skip_contents && transfer_contents)
+		N.contents = O.contents
+		if(istype(N,/obj/item/weapon/gun/projectile/))
+			N:magazine_type = O:magazine_type
+			N:ammo_magazine = O:ammo_magazine
+		if(istype(N,/obj/item/weapon/gun/energy/))
+			N:cell_type = O:cell_type
+	else	//nuke any ammo it'd normally spawn with, if it's a gun, to prevent ammo duplication. we have to do this just for guns, not immediately under the else, or it breaks default attachments like hoods and suit storage
+		if(istype(N,/obj/item/weapon/gun/projectile/))
+			N:contents = list()
+			N:magazine_type = null
+			N:ammo_magazine = null
+		if(istype(N,/obj/item/weapon/gun/energy/))
+			N:contents = list()
+			N:cell_type = null
+	
```

Comment:
Stuff like this might be worth having a `obj/item/proc/customize_item(var/obj/item/new_item)` so you can just have each object do what it wants rather than trying to handle the special cases all here.

Reply:
Maybe? I figure the way kits are set up now covers all the major bases though. You basically have to set the kit up (or varedit it) in a specifically unfriendly way in order to break it. I think.

Honestly I just kinda bash things together until they work, and then I'm pleasantly surprised when they do.

The whole issue essentially stems from the fact YW doesn't allow people to spawn with custom weapons/permits. That means modifying existing station equipment and _that_ means putting your fluff weapon back into storage when you leave the round where other people can get their grubby hands all over it, which some people might not be comfortable with. I set out to simply permit reverting changes initially, but then I realized that you could do stupid things with them if you were careless (like accidentally delete items attached to voidsuits or ammo loaded into guns), and it also basically let you do a bunch of other abusive stuff like duping ammo (if guns spawn loaded, and 99% of them do) or trivially removing forensic evidence of crimes if you were suitably conniving... so here we are.